### PR TITLE
Add separate section for array functions, add strcat-array

### PR DESCRIPTION
**Pull request by manototh**
Preview:

- [Overview](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions)
- [array_concat](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-concat)
- [array_iff](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-iff)
- [array_index_of](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-index-of)
- [array_length](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-length)
- [array_reverse](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-reverse)
- [array_rotate_left](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-rotate-left)
- [array_rotate_right](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-rotate-right)
- [array_select_dict](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-select-dict)
- [array_shift_left](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-shift-left)
- [array_shift_right](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-shift-right)
- [array_slice](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-slice)
- [array_split](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-split)
- [array_sum](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/array-sum)
- [isarray](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/isarray)
- [pack_array](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/pack-array)
- [strcat_array](https://axiom-mano-array-functions.mintlify.app/apl/scalar-functions/array-functions/strcat-array)

### CODE REVIEW

File: apl/scalar-functions/array-functions.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: 'Array functions'
 description: 'Learn how to use and combine different mathematical functions in APL'
```
Comment (ted-gould) [inline apl/scalar-functions/array-functions.mdx]: Description seems wrong here.

File: apl/scalar-functions/array-functions/array-concat.mdx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+In SPL, you typically use the `mvappend` function to concatenate multiple fields or arrays into a single array. In APL, the equivalent is `array_concat`, which also combines arrays but requires you to specify each array as a parameter.
+
+<CodeGroup>
+```sql Splunk example
+| eval combined_array = mvappend(array1, array2, array3)
+```
+
+```kusto APL equivalent
+let combined_array = array_concat(array1, array2, array3)
```
Comment (ted-gould) [inline apl/scalar-functions/array-functions/array-concat.mdx]: Not sure we want to use `let` here? Perhaps `| extend` ?

File: apl/scalar-functions/array-functions/array-iff.mdx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+<Tab title="Log analysis">
+
+The `array_iff` function can help filter log data conditionally, such as choosing specific durations based on HTTP status codes.
+
+**Query**
+
+```kusto
+['sample-http-logs']
+| extend duration_conditions = array_iff(status == '200', req_duration_ms, 0)
+| project _time, duration_conditions
```
Comment (ted-gould) [inline apl/scalar-functions/array-functions/array-iff.mdx]: How about something like this: https://play.axiom.co/axiom-play-qf1k/explorer?qid=pZxTWjaWAAc-sm6fpl&relative=1

File: apl/scalar-functions/array-functions/array-index-of.mdx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+<Tabs>
+<Tab title="Log analysis">
+
+You can use `array_index_of` to find the position of a specific HTTP status code within an array of codes in your log analysis.
+
+**Query**
+
+```kusto
+['sample-http-logs']
+| extend status_array = make_list(status)
```
Comment (ted-gould) [inline apl/scalar-functions/array-functions/array-index-of.mdx]: Make list has to be in a summarize

File: apl/scalar-functions/array-functions.mdx [hunk 5]
```diff
@@ -1,613 +1,34 @@
...
+| [array_reverse](/apl/scalar-functions/array-functions/array-reverse)         | Reverses the order of the elements in a dynamic array.                                                              |
+| [array_rotate_left](/apl/scalar-functions/array-functions/array-rotate-left)   | Rotates values inside a dynamic array to the left.                                                                |
+| [array_rotate_right](/apl/scalar-functions/array-functions/array-rotate-right) | Rotates values inside a dynamic array to the right.                                                               |
+| [array_select_dict](/apl/scalar-functions/array-functions/array-select-dict) | Selects a dictionary from an array of dictionaries.                                                                 |
+| [array_shift_left](/apl/scalar-functions/array-functions/array-shift-left)   | Shifts the values inside a dynamic array to the left.                                                             |
+| [array_shift_right](/apl/scalar-functions/array-functions/array-shift-right) | shifts values inside an array to the right.                                                                         |
```
Comment (ted-gould) [inline apl/scalar-functions/array-functions.mdx]: `shifts` should be capitalized

File: apl/scalar-functions/array-functions.mdx [hunk 6]
```diff
@@ -1,613 +1,34 @@
...
+| [array_split](/apl/scalar-functions/array-functions/array-split)             | Splits an array to multiple arrays according to the split indices and packs the generated array in a dynamic array. |
+| [array_sum](/apl/scalar-functions/array-functions/array-sum)                 | Calculates the sum of elements in a dynamic array.                                                                  |
+| [isarray](/apl/scalar-functions/array-functions/isarray)                     | Checks whether a value is an array.                                                                                 |
+| [pack_array](/apl/scalar-functions/array-functions/pack-array)               | Packs all input values into a dynamic array.                                                                        |
+| [strcat-array](/apl/scalar-functions/array-functions/strcat-array)               | Takes an array and returns a single concatenated string with the array’s elements separated by the specified delimiter. |
```
Comment (ted-gould) [inline apl/scalar-functions/array-functions.mdx]: Should be `strcat_array`

File: apl/scalar-functions/array-functions/array-iff.mdx [hunk 7]
```diff
@@ -97,18 +98,19 @@ With OpenTelemetry trace data, you can use `array_iff` to filter spans based on
...
 
 **Output**
 
-| _time              | service.name       | server_durations |
-|--------------------|--------------------|------------------|
-| 2023-10-18T08:50Z | frontend           | [30ms, 0ms]     |
-| 2023-10-18T09:20Z | cartservice        | [0ms, 50ms]     |
+| server_durations |
+| --- |
+| ["45.632µs", "54.622µs", 0, "34.051µs"] |
```
Comment (ted-gould) [inline apl/scalar-functions/array-functions/array-iff.mdx:111]: I think this implies they'll be string, which they won't. I think leave the quotes off.

**Review by ted-gould — APPROVED** [review]
Looks good, just a couple little things I noticed.

File: apl/scalar-functions/array-functions/array-iff.mdx (hunk 7 above)
Comment (manototh) [reply apl/scalar-functions/array-functions/array-iff.mdx:111]: Thanks for noticing this. This is how they show up in the results: https://play.axiom.co/axiom-play-qf1k/explorer?qid=Ih31DhodZdm-sofe17&relative=1
I would leave them with quotation marks.

**Pull request merged**